test/io/ferguson/read-class-reorder.chpl

```diff
@@ -0,0 +1,29 @@
+/*class Base {
+}
+
+class Parent : Base {
+  var x: int;
+  var y: int;
+}
+*/
+
+class Child /*: Parent*/ {
```

Comment:
Yeah, but that one only covered inherited fields.  I was thinking of an overlap test - inherited fields plus assignment in different order.